salt/modules/mac_service.py

```diff
@@ -77,8 +77,50 @@ def _get_service(name):
             # Match on basename
             return service
 
-    # Could not find service
-    raise CommandExecutionError('Service not found: {0}'.format(name))
+    return dict()
+
+
+def _get_service(name):
+    '''
+    Get information about a service.  If the service is not found, raise an
+    error
+
+    :param str name: Service label, file name, or full path
+
+    :return: The service information for the service, otherwise an Error
+    :rtype: dict
+    '''
+    services = __utils__['mac_utils.available_services']()
+    name = name.lower()
+
+    service = _name_in_services(name, services)
+
+    # if we would the service we can return it
+    if service:
+        return service
+
+    # if we got here our service is not available, now we can check to see if
+    # we received a cached batch of services, if not we did a fresh check
+    # so we need to raise that the service could not be found.
+    try:
+        if not __context__['using_cached_services']:
```

Comment:
i'm not seeing `using_cached_services` anywhere else in the code. Did you mean for that to be `available_services_cached` possibly or maybe i'm not reading it correctly?

Reply:
Good catch, I changed it to be somewhat more human readable and missed that one.

Reply:
updated with the most recent commit.